userdocs/creating-a-codelist-from-a-csv-file.md

```diff
@@ -5,15 +5,22 @@ As well as creating a codelist from scratch, you can create one by uploading a C
 From the _My codelists_ page, click _Create a codelist_.
 
 Choose a name for the codelist, select a coding system from the dropdown,
-and choose a file to upload from your hard drive.  If you are a member of an organisation, you can also choose an owner for the codelist (your own account or an organisation account).
+and choose a file to upload from your hard drive.  If you wish to create an OPCS-4 codelist, please see the notes at the bottom of this page. If you are a member of an organisation, you can also choose an owner for the codelist (your own account or an organisation account).
 
         create-a-codelist-1.png
 
-Note that the CSV file should not have a header,
-and its first column must contain valid codes in the chosen coding system.
-(This limitation will be lifted.)
+The final codelist must be stored in CSV format to be imported 
+into [OpenCodelists](https://www.opencodelists.org). The codes must be stored in exactly one column.
+There is currently a soft requirement that the first column must contain codes in the chosen coding system, preferably named according to the CSV column names provided in the table above (this limitation will be lifted). The second column is typically a text description of the code. Before uploading a codelist CSV under your OpenSAFELY codelist profile, you should remove the header row as these will be automatically added in a standardised manner.
+
+The codelist page will allow you to upload two columns (a code, and a text description of the code), however, some codelists may require a 'classification' or 'type' column, which classifies the codes into subcategories. For example, when using a codelist for venous thromboemobolism you may wish to classify these codes into deep vein thromboses and pulmonary embolisms, and keep this within a single codelist rather than uploading separate lists for each subcategory of the clinical condition. You can access subcategories of a codelist by using the `filter_codes_by_category` functionality when calling `with_these_clinical_events` as part of your `study_definition`. Uploading more than two columns is currently only possible for the OpenSAFELY core team, so in case this is required for your study please get in touch.
+
+Finally, please avoid filtering on an include or exclude column in Excel when finalising a codelist. Any filters applied will be lost when converted to CSV, and you will end up with all of the codes being uploaded. Please also avoid opening any SNOMED-CT codelists in Excel, as the codes get rounded.
```

Comment:
```suggestion
### More about codelist columns

The codelist page allow you to upload two columns:

* a code
* a text description of the code

However, some codelists may require a 'classification' or 'type' column, which classifies the codes into subcategories. For example, when using a codelist for venous thromboembolism, you may wish to classify these codes into deep vein thromboses and pulmonary embolisms. By using subcategories, you can keep all the codes in a single codelist, rather than uploading separate lists for each clinical subcategory. In your study definition, the `filter_codes_by_category` functionality allows access to the subcategories of a codelist.
 
**Uploading more than two columns is currently only possible for the OpenSAFELY core team. If your study requires this feature, [please get in touch](https://www.opensafely.org/contact/).**

### Potential issues when editing codelists in spreadsheet software, such as Excel

Avoid:

* filtering on an include or exclude column when finalising a codelist. Applied filters are lost in CSV conversion: *all* of the codes will be uploaded.
* editing SNOMED-CT codelists. The codes get rounded.
```

* I've referred the reader to the OpenSAFELY contact page as the call to action, because that's listed on the landing page of Opencodelists.
* I hope I've got the gist of the spreadsheet gotchas. If my edits have missed something subtle, feel free to tweak that more. Likewise with anything else here, I might have confused things more than helped!